### PR TITLE
fix: notifications: use username in workspace URLs

### LINKED ISSUE #13998: bug: broken workspace URL in notifications
Apparently, notifications link to invalid username handle.

Screenshot 2024-07-24 at 11 46 37

Screenshot 2024-07-24 at 11 46 30

Currently:

`https://dev.coder.com/@Marcin%20Tojek/workspaceworkspaceworkspaceworks`

but should be:

`https://dev.coder.com/@mtojek/workspaceworkspaceworkspaceworks`

### PR DESCRIPTION
**Pull request by mtojek**
Fixes: https://github.com/coder/coder/issues/13998

This PR adjusts notification templates to use correct username handle in notification actions.

### CODE REVIEW

File: coderd/database/queries/notifications.sql [hunk 1]
```diff
@@ -4,7 +4,8 @@ SELECT nt.name                                                    AS notificatio
        nt.actions                                                 AS actions,
        u.id                                                       AS user_id,
        u.email                                                    AS user_email,
-       COALESCE(NULLIF(u.name, ''), NULLIF(u.username, ''))::text AS user_name
+       COALESCE(u.name, '')                                       AS user_name,
+       COALESCE(u.username, '')                                   AS user_username
```
Comment (dannykopping) [inline coderd/database/queries/notifications.sql:8]: We should fall back to `u.username` if `u.name` is not set.

```suggestion
       COALESCE(NULLIF(u.name, ''), NULLIF(u.username, ''))::text    AS user_name,
       u.username                                                                                        AS user_username
```

File: coderd/database/migrations/000230_notifications_fix_username.down.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+UPDATE notification_templates
+SET
+    actions = REPLACE(actions::text, '@{{.UserUsername}}', '@{{.UserName}}')::jsonb;
```
Comment (dannykopping) [inline coderd/database/migrations/000230_notifications_fix_username.down.sql:3]: We can't replace this everywhere; when we greet the user we should use `UserName`, but when we reference their account we should use `UserUsername`.

Comment (mtojek) [reply coderd/database/migrations/000230_notifications_fix_username.down.sql:3]: It replaces values only in the actions column, right?

File: coderd/database/queries/notifications.sql (hunk 1 above)
Comment (mtojek) [reply coderd/database/queries/notifications.sql:8]: Is it a valid case that `Name` is not set?

Comment (dannykopping) [reply coderd/database/queries/notifications.sql:8]: Yup, it's a new field I believe.

File: coderd/database/migrations/000230_notifications_fix_username.down.sql (hunk 2 above)
Comment (dannykopping) [reply coderd/database/migrations/000230_notifications_fix_username.down.sql:3]: 🤦 missed that 👍

File: coderd/database/queries/notifications.sql (hunk 1 above)
Comment (mtojek) [reply coderd/database/queries/notifications.sql:8]: Let me quickly patch it.

Comment (mtojek) [reply coderd/database/queries/notifications.sql]: fixed

**Review by dannykopping — APPROVED** [review]
LGTM

**Review by dannykopping — APPROVED** [review]
LGTM

**Pull request merged**